src/lib/RpcServer.es.js

```diff
@@ -12,258 +11,138 @@
-    ExtraJSONTypes[ExtraJSONTypes["UINT8_ARRAY"] = 0] = "UINT8_ARRAY";
-})(ExtraJSONTypes || (ExtraJSONTypes = {}));
-class JSONUtils {
-    static stringify(value) {
-        return JSON.stringify(value, JSONUtils._jsonifyType);
-    }
-    static parse(value) {
-        return JSON.parse(value, JSONUtils._parseType);
-    }
-    static _parseType(key, value) {
-        if (value && value.hasOwnProperty &&
-            value.hasOwnProperty(JSONUtils.TYPE_SYMBOL) && value.hasOwnProperty(JSONUtils.VALUE_SYMBOL)) {
-            switch (value[JSONUtils.TYPE_SYMBOL]) {
-                case ExtraJSONTypes.UINT8_ARRAY:
-                    return Base64.decode(value[JSONUtils.VALUE_SYMBOL]);
-            }
-        }
-        return value;
-    }
-    static _jsonifyType(key, value) {
-        if (value instanceof Uint8Array) {
-            return JSONUtils._typedObject(ExtraJSONTypes.UINT8_ARRAY, Base64.encode(value));
-        }
-        return value;
-    }
-    static _typedObject(type, value) {
-        const obj = {};
-        obj[JSONUtils.TYPE_SYMBOL] = type;
-        obj[JSONUtils.VALUE_SYMBOL] = value;
-        return obj;
-    }
-}
-JSONUtils.TYPE_SYMBOL = '__';
-JSONUtils.VALUE_SYMBOL = 'v';
+const ResponseStatus = {
+    OK: 'ok',
+    ERROR: 'error',
+};
 
 class UrlRpcEncoder {
+    /**
+     * @param {Location} url
+     * @returns {Message?}
+     */
     static receiveRedirectCommand(url) {
         // Need referrer for origin check
-        if (!document.referrer)
-            return null;
+        if (!document.referrer) return null;
         // Parse query
         const params = new URLSearchParams(url.search);
         const referrer = new URL(document.referrer);
         // Ignore messages without a command
-        if (!params.has('command'))
-            return null;
+        if (!params.has('command')) return null;
         // Ignore messages without an ID
-        if (!params.has('id'))
-            return null;
+        if (!params.has('id')) return null;
         // Ignore messages without a valid return path
-        if (!params.has('returnURL'))
-            return null;
+        if (!params.has('returnURL')) return null;
         // Only allow returning to same origin
-        const returnURL = new URL(params.get('returnURL'));
-        if (returnURL.origin !== referrer.origin)
-            return null;
+        const returnURL = new URL(/** @type {string} */ (params.get('returnURL')));
+        if (returnURL.origin !== referrer.origin) return null;
         // Parse args
         let args = [];
         if (params.has('args')) {
             try {
-                args = JSONUtils.parse(params.get('args'));
-            }
-            catch (e) {
+                args = JsonUtils.parse(/** @type {string} */ (params.get('args')));
+            } catch (e) {
                 // Do nothing
             }
         }
         args = Array.isArray(args) ? args : [];
-        return {
+
+        return ({
```

Comment:
```suggestion
        return {
```
Probably a leftover from an annotation.